include/nbl/builtin/hlsl/shared_memory_accessor.hlsl

```diff
@@ -0,0 +1,155 @@
+    void get(const uint ix, out float3 value) { value = asfloat(uint3(accessor.get(ix), accessor.get(ix + _NBL_HLSL_WORKGROUP_SIZE_), accessor.get(ix + 2 * _NBL_HLSL_WORKGROUP_SIZE_)));}
+    void get(const uint ix, out float4 value) { value = asfloat(uint4(accessor.get(ix), accessor.get(ix + _NBL_HLSL_WORKGROUP_SIZE_), accessor.get(ix + 2 * _NBL_HLSL_WORKGROUP_SIZE_), accessor.get(ix + 3 * _NBL_HLSL_WORKGROUP_SIZE_)));}
+
+	void set(const uint ix, const uint value) {accessor.set(ix, value);}
+    void set(const uint ix, const uint2 value) {
+        accessor.set(ix, value.x);
+        accessor.set(ix + _NBL_HLSL_WORKGROUP_SIZE_, value.y);
+    }
+    void set(const uint ix, const uint3 value)  {
+        accessor.set(ix, value.x);
+        accessor.set(ix + _NBL_HLSL_WORKGROUP_SIZE_, value.y);
+        accessor.set(ix + 2 * _NBL_HLSL_WORKGROUP_SIZE_, value.z);
+    }
+    void set(const uint ix, const uint4 value) {
+        accessor.set(ix, value.x);
+        accessor.set(ix + _NBL_HLSL_WORKGROUP_SIZE_, value.y);
+        accessor.set(ix + 2 * _NBL_HLSL_WORKGROUP_SIZE_, value.z);
+        accessor.set(ix + 3 * _NBL_HLSL_WORKGROUP_SIZE_, value.w);
+    }
+
+    void set(const uint ix, const int value) {accessor.set(ix, asuint(value));}
+    void set(const uint ix, const int2 value) {
+        accessor.set(ix, asuint(value.x));
+        accessor.set(ix + _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.y));
+    }
+    void set(const uint ix, const int3 value)  {
+        accessor.set(ix, asuint(value.x));
+        accessor.set(ix + _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.y));
+        accessor.set(ix + 2 * _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.z));
+    }
+    void set(const uint ix, const int4 value) {
+        accessor.set(ix, asuint(value.x));
+        accessor.set(ix + _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.y));
+        accessor.set(ix + 2 * _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.z));
+        accessor.set(ix + 3 * _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.w));
+    }
+
+    void set(const uint ix, const float value) {accessor.set(ix, asuint(value));}
+    void set(const uint ix, const float2 value) {
+        accessor.set(ix, asuint(value.x));
+        accessor.set(ix + _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.y));
+    }
+    void set(const uint ix, const float3 value)  {
+        accessor.set(ix, asuint(value.x));
+        accessor.set(ix + _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.y));
+        accessor.set(ix + 2 * _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.z));
+    }
+    void set(const uint ix, const float4 value) {
+        accessor.set(ix, asuint(value.x));
+        accessor.set(ix + _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.y));
+        accessor.set(ix + 2 * _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.z));
+        accessor.set(ix + 3 * _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.w));
+    }
+    
+    // REVIEW:  Depending on how we handle SHARED_MEM in nbl::hlsl::atomics 
+	//			we should also update this part to handle float and int
```

Comment:
by all means go ahead with `int` and `float` via casting to `uint`

But DO NOT TRY TO IMPLEMENT `Add`,`Min`, `Max`, CompSwap`